protocols/virtual-fund/virtual-fund.go

```diff
@@ -59,12 +59,23 @@ func New(
 	ledgerChannelToMyRight channel.Channel,
 ) (VirtualFundObjective, error) {
 
-	var init VirtualFundObjective
-
-	if !(ledgerChannelToMyLeft.IsTwoPartyLedger && ledgerChannelToMyRight.IsTwoPartyLedger) {
-		return init, errors.New(`supplied channels are not two party ledger channels`)
+	switch myRole {
+	case 0: // Alice
+		if !ledgerChannelToMyRight.IsTwoPartyLedger {
+			return VirtualFundObjective{}, errors.New(`alice's right-channel is not a two-party ledger channel`)
+		}
+	case n + 1: // Bob
```

Comment:
Is there an off-by-one error here? Should this be `n`?

Reply:
An excellent and prudent question, but: nope - this is per the updated SATP paper. `n` here refers to n intermediaries - 0 = Alice, 1, ... , `n` the intermediaries, `n+1` = Bob